llumnix/backends/vllm/scheduler.py

```diff
@@ -205,8 +205,10 @@ def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
         self.update_instance_info_callback(self._get_instance_info())
         return seq_group_metadata_list, scheduler_outputs
 
-    @scheduler_lock
     def add_seq_group(self, *args, **kwargs):
+        # The scheduler lock is mannually released in the end of LLMEngineLlumnix.add_request function.
```

Comment:
why?

Reply:
Such design looks dangerous. Better to avoid it

Reply:
Because we need to write the waiting queue of the scheduler in LLMEngineLlumnix.add_request. Since we will upgrade vllm and use asynchronous actor rather than threaded actor in the near future, the scheduler lock will no longer be needed after that, so we apply this hack way temporarily.